tests/client_test.go

```diff
@@ -992,13 +1144,23 @@ func TestUpstreamServerWithDrain(t *testing.T) {
 		t.Fatalf("Error connecting to nginx: %v", err)
 	}
 
+	maxConns := 0
```

Comment:
in case we're setting the default values should we use global variables like `defaultMaxFails` and `defaultWeight` for consistency?
here and in `TestUpstreamServerDefaultParameters` and  `TestStreamUpstreamServerDefaultParameters`